sleap/nn/inference.py

```diff
@@ -5461,43 +5490,169 @@ def main(args: Optional[list] = None):
+                output_path = (
+                    output_path_obj.as_posix()
+                    + "/"
+                    + (data_path_obj.stem + ".predictions.slp")
+                )
+                output_path_obj = Path(output_path)
+
+            labels_pr.provenance["model_paths"] = predictor.model_paths
+            labels_pr.provenance["predictor"] = type(predictor).__name__
+
+            if args.no_empty_frames:
+                # Clear empty frames if specified.
+                labels_pr.remove_empty_frames()
+
+            finish_timestamp = str(datetime.now())
+            total_elapsed = time() - t0
+            print("Finished inference at:", finish_timestamp)
+            print(f"Total runtime: {total_elapsed} secs")
+            print(f"Predicted frames: {len(labels_pr)}/{len(provider)}")
+
+            # Add provenance metadata to predictions.
+            labels_pr.provenance["sleap_version"] = sleap.__version__
+            labels_pr.provenance["platform"] = platform.platform()
+            labels_pr.provenance["command"] = " ".join(sys.argv)
+            labels_pr.provenance["data_path"] = data_path_obj.as_posix()
+            labels_pr.provenance["output_path"] = output_path_obj.as_posix()
+            labels_pr.provenance["total_elapsed"] = total_elapsed
+            labels_pr.provenance["start_timestamp"] = start_timestamp
+            labels_pr.provenance["finish_timestamp"] = finish_timestamp
+
+            print("Provenance:")
+            print(labels_pr.provenance)
+            print()
 
+            labels_pr.provenance["args"] = vars(args)
+
+            # Save results.
+            labels_pr.save(output_path)
+            print("Saved output:", output_path)
+
+            if args.open_in_gui:
+                subprocess.call(["sleap-label", output_path])
+
+            # Reset output_path for next iteration
+            output_path = args.output
+
+    # running tracking on existing prediction file
     elif getattr(args, "tracking.tracker") is not None:
-        # Load predictions
-        print("Loading predictions...")
-        labels_pr = sleap.load_file(args.data_path)
-        frames = sorted(labels_pr.labeled_frames, key=lambda lf: lf.frame_idx)
+        for data_path, provider in zip(data_path_list, provider_list):
+            # Load predictions
+            data_path_obj = Path(data_path)
+            print("Loading predictions...")
+            labels_pr = sleap.load_file(data_path_obj.as_posix())
+            frames = sorted(labels_pr.labeled_frames, key=lambda lf: lf.frame_idx)
+
+            print("Starting tracker...")
+            frames = run_tracker(frames=frames, tracker=tracker)
+            tracker.final_pass(frames)
+
+            labels_pr = Labels(labeled_frames=frames)
+
+            if output_path is None:
+                output_path = f"{data_path}.{tracker.get_name()}.slp"
+                output_path_obj = Path(output_path)
+
+            if output_path is not None:
```

Comment:
Probably don't need this check right? By this line, `output_path` is never `None` since it's set to a path in the previous couple of lines if it was `None`.